src/main/java/org/opensearch/security/filter/SecurityRestFilter.java

```diff
@@ -145,7 +151,56 @@ private boolean userIsSuperAdmin(User user, AdminDNs adminDNs) {
         return user != null && adminDNs.isAdmin(user);
     }
 
-    private boolean checkAndAuthenticateRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+    private boolean authorizeRequest(RestHandler original, RestRequest request, RestChannel channel, User user) {
```

Comment:
Shouldn't this return a `PrivilegesEvaluatorResponse` so it can be passed into the rest of the ecosystem, such as in the transport layer?